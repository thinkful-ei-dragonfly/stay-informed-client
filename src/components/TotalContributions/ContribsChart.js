import React from "react";
import Chart from "react-google-charts";


class ContribsChart extends React.Component {

  donations = parseFloat(this.props.donations);
  spent = parseFloat(this.props.spent);

  data = [
    ["", "$", { role: "style" }],
    ["Total Donations", this.donations, "color: green"],
    ["Total Spent", this.spent, "color: red"],
  ];

  options={
    title: 'Contributions in the past cycle',
    chartArea: { width: '50%' },
    hAxis: {
      title: 'Dollars ($)',
      minValue: 0,
    },
    vAxis: {
      title: 'City',
    },
  }
 
  render() {
    return (
      <div className="chart">
        <Chart chartType="BarChart" width="100%" height="200px" data={this.data} options={this.options}/>
      </div>
    );
  }
}

export default ContribsChart;