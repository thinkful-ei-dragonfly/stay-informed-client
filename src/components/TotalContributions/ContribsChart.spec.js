import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ContribsChart from './ContribsChart'

describe(`ContribsChart component`, () => {
  it('renders a form.ContribsChart by default', () => {
    const wrapper = shallow(<ContribsChart />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
