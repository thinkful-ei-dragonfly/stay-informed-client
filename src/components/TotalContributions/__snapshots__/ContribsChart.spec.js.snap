// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContribsChart component renders a form.ContribsChart by default 1`] = `
<div
  className="chart"
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="BarChart"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          "",
          "$",
          Object {
            "role": "style",
          },
        ],
        Array [
          "Total Donations",
          NaN,
          "color: green",
        ],
        Array [
          "Total Spent",
          NaN,
          "color: red",
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="200px"
    legendToggle={false}
    legend_toggle={false}
    options={
      Object {
        "chartArea": Object {
          "width": "50%",
        },
        "hAxis": Object {
          "minValue": 0,
          "title": "Dollars ($)",
        },
        "title": "Contributions in the past cycle",
        "vAxis": Object {
          "title": "City",
        },
      }
    }
    render={null}
    rootProps={Object {}}
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="100%"
  />
</div>
`;
